src/crunch.rs

```diff
@@ -0,0 +1,68 @@
+pub(crate) mod crn_utils;
+pub(crate) mod crn_consts;
+pub(crate) mod crn_static_huffman_data_model;
+pub(crate) mod crn_symbol_codec;
+pub(crate) mod crn_unpacker;
+pub(crate) mod crn_decomp;
+use super::crnlib::{CrnTextureInfo, CrnFormat};
+use core::cmp::max;
+use crate::bcn;
+extern crate alloc;
+
+struct CrunchDecodeHandler{
+    format: CrnFormat,
+    dxt_data: alloc::vec::Vec<u8>
+}
+
+fn crunch_unpack_level<'vec>(data: &[u8], data_size: u32, level_index: u32) -> Result<CrunchDecodeHandler, &'static str> {
+    let mut tex_info: CrnTextureInfo = CrnTextureInfo::default();
+    if tex_info.crnd_get_texture_info(data, data_size) == false {
+        return Err("Invalid crunch texture encoding.");

```

Comment:
In Rust, the convention is to have error messages in lowercase and with no trailing punctuation
```suggestion
        return Err("invalid crunch texture encoding");

```

Reply:
same goes for all other messages